lib/puppet/util/execution.rb

```diff
@@ -187,18 +187,30 @@ def self.execute(command, options = NoOptionsSpecified)
     null_file = Puppet.features.microsoft_windows? ? 'NUL' : '/dev/null'
 
     begin
+      reader, writer = IO.pipe unless options[:squelch]
```

Comment:
I'm not sure what to do about that. It's not ideal to be unable to run Puppet again if a previous run was killed but left a long-running or hung process behind, but I'm not sure when that would really come up.